src/sentry/api/endpoints/organization_transaction_details.py

```diff
@@ -0,0 +1,314 @@
+from collections import defaultdict
+from datetime import datetime, timedelta, timezone
+
+from rest_framework.request import Request
+from rest_framework.response import Response
+from sentry_relay.consts import SPAN_STATUS_CODE_TO_NAME
+from snuba_sdk import Column
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.constants import ObjectStatus
+from sentry.models.project import Project
+from sentry.search.events.builder.spans_indexed import SpansIndexedQueryBuilder
+from sentry.search.events.types import ParamsType
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+
+
+@region_silo_endpoint
+class OrganizationTransactionDetailsEndpoint(OrganizationEndpoint):
+    owner = ApiOwner.PERFORMANCE
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+
+    def get(self, request: Request, organization, project_slug, transaction_id):
+        try:
+            project = Project.objects.get(
+                slug=project_slug, organization_id=organization.id, status=ObjectStatus.ACTIVE
+            )
+        except Project.DoesNotExist:
+            return Response(status=404)
+
+        # Check access to the project as this endpoint doesn't use membership checks done
+        # get_filter_params().
+        if not request.access.has_project_access(project):
+            return Response(status=404)
+
+        # Get a list of all spans with this transaction_id
+        start = request.GET.get("start_timestamp")
+        end = request.GET.get("end_timestamp")
+
+        spans_data = _query_all_spans_in_transaction(
+            organization, project, transaction_id, start, end
```

Comment:
I'm calling it `transaction_id` in the endpoint params because "transaction" is the name the frontend uses, and "segment" is never exposed - right? That's my understanding, anyway. (Otherwise this should be a "segment details" endpoint instead, right?). I could be misunderstanding though!